Cesium3DTilesSelection/src/Tile.cpp

```diff
@@ -766,111 +863,12 @@ void Tile::update(
-      }
-      if (this->_pContent->updatedContentBoundingVolume) {
-        // There should not be an updated content bounding volume unless there
-        // is also an updated regular bounding volume.
-        assert(this->_pContent->updatedBoundingVolume);
-        this->setContentBoundingVolume(
-            *this->_pContent->updatedContentBoundingVolume);
-      }
-
-      if (!this->_pContent->availableTileRectangles.empty() &&
-          this->getContext()->implicitContext) {
-        ImplicitTilingContext& context =
-            this->getContext()->implicitContext.value();
-        for (const QuadtreeTileRectangularRange& range :
-             this->_pContent->availableTileRectangles) {
-          context.availability.addAvailableTileRange(range);
-        }
-      }
-    }
-
-    this->setState(LoadState::Done);
-  }
-
-  if (this->getContext()->implicitContext && this->getChildren().empty() &&
-      std::get_if<QuadtreeTileID>(&this->_id)) {
-    // Check if any child tiles are known to be available, and create them if
-    // they are.
-    const ImplicitTilingContext& implicitContext =
-        this->getContext()->implicitContext.value();
-    const CesiumGeometry::QuadtreeTileAvailability& availability =
-        implicitContext.availability;
-
-    const QuadtreeTileID id = std::get<QuadtreeTileID>(this->_id);
-
-    const QuadtreeTileID swID(id.level + 1, id.x * 2, id.y * 2);
-    const uint32_t sw = availability.isTileAvailable(swID) ? 1 : 0;
-
-    const QuadtreeTileID seID(swID.level, swID.x + 1, swID.y);
-    const uint32_t se = availability.isTileAvailable(seID) ? 1 : 0;
-
-    const QuadtreeTileID nwID(swID.level, swID.x, swID.y + 1);
-    const uint32_t nw = availability.isTileAvailable(nwID) ? 1 : 0;
-
-    const QuadtreeTileID neID(swID.level, swID.x + 1, swID.y + 1);
-    const uint32_t ne = availability.isTileAvailable(neID) ? 1 : 0;
-
-    const size_t childCount = sw + se + nw + ne;
-    if (childCount > 0) {
-      // If any children are available, we need to create all four in order to
-      // avoid holes. But non-available tiles will be upsampled instead of
-      // loaded.
-      // TODO: this is the right thing to do for terrain, which is the only use
-      // of implicit tiling currently. But we may need to re-evaluate it if
-      // we're using implicit tiling for buildings (for example) in the future.
-      this->_children.resize(4);
-
-      createImplicitTile(implicitContext, *this, this->_children[0], swID, sw);
-      createImplicitTile(implicitContext, *this, this->_children[1], seID, se);
-      createImplicitTile(implicitContext, *this, this->_children[2], nwID, nw);
-      createImplicitTile(implicitContext, *this, this->_children[3], neID, ne);
-    }
+  std::optional<CesiumGeospatial::GlobeRectangle> rectangle =
+      estimateGlobeRectangle(this->getBoundingVolume());
+  if (!rectangle) {
+    return;
   }
 
-  // TODO: if there's no model, we can actually free any existing overlays.
```

Comment:
This TODO is still relevant, I think. Though it should probably become an issue rather than a TODO.